--- origsrc/gwenhywfar-3.2.0/plugins/ct/ohbci/Makefile.am	2007-11-22 12:27:15.000000000 -0600
+++ src/gwenhywfar-3.2.0/plugins/ct/ohbci/Makefile.am	2008-03-28 06:07:55.687500000 -0500
@@ -29,7 +29,7 @@
 # IS_WINDOWS
 
 ohbci_la_LDFLAGS = -no-undefined $(STRIPALL) $(ressource_ldflags) \
-  -module -omit-version
+  -module -avoid-version
 
 sources:
 	for f in $(ohbci_la_SOURCES); do \
--- origsrc/gwenhywfar-3.2.0/plugins/dbio/csv/Makefile.am	2007-11-22 12:27:15.000000000 -0600
+++ src/gwenhywfar-3.2.0/plugins/dbio/csv/Makefile.am	2008-03-28 06:06:29.734375000 -0500
@@ -28,7 +28,7 @@
 # IS_WINDOWS
 
 csv_la_LDFLAGS = -no-undefined $(STRIPALL) $(ressource_ldflags) \
-    -module -version-info 0:1:0
+    -module -avoid-version
 
 
 sources:
--- origsrc/gwenhywfar-3.2.0/plugins/dbio/olddb/Makefile.am	2007-11-22 12:27:15.000000000 -0600
+++ src/gwenhywfar-3.2.0/plugins/dbio/olddb/Makefile.am	2008-03-28 06:06:34.265625000 -0500
@@ -28,7 +28,7 @@
 # IS_WINDOWS
 
 olddb_la_LDFLAGS = -no-undefined $(STRIPALL) $(ressource_ldflags) \
-    -module -version-info 0:1:0
+    -module -avoid-version
 
 sources:
 	for f in $(olddb_la_SOURCES); do \
--- origsrc/gwenhywfar-3.2.0/plugins/dbio/xmldb/Makefile.am	2007-11-22 12:27:15.000000000 -0600
+++ src/gwenhywfar-3.2.0/plugins/dbio/xmldb/Makefile.am	2008-03-28 06:06:42.156250000 -0500
@@ -28,7 +28,7 @@
 # IS_WINDOWS
 
 xmldb_la_LDFLAGS = -no-undefined $(STRIPALL) $(ressource_ldflags) \
-    -module -version-info 0:1:0
+    -module -avoid-version
 
 sources:
 	for f in $(xmldb_la_SOURCES); do \
