--- origsrc/gwenhywfar-4.13.1/src/sio/syncio_tls.c	2014-10-12 08:12:22.000000000 -0500
+++ src/gwenhywfar-4.13.1/src/sio/syncio_tls.c	2015-10-16 12:14:43.337776700 -0500
@@ -350,7 +350,7 @@ int GWEN_SyncIo_Tls_Prepare(GWEN_SYNCIO
   }
 
   /* set default priority */
-#if 1 /* old code */
+#if 0 /* old code */
   GWEN_Gui_ProgressLog(0, GWEN_LoggerLevel_Error, I18N("Using old SSL preparation code."));
   rv=gnutls_set_default_priority(xio->session);
   if (rv) {
