--- origsrc/gwenhywfar-4.13.1/src/sio/syncio_tls.c	2014-10-12 08:12:22.000000000 -0500
+++ src/gwenhywfar-4.13.1/src/sio/syncio_tls.c	2015-10-16 12:14:43.337776700 -0500
@@ -490,7 +490,6 @@ int GWEN_SyncIo_Tls_Prepare(GWEN_SYNCIO
   /* find default trust file if none is selected */
   if (lflags & GWEN_SYNCIO_TLS_FLAGS_ADD_TRUSTED_CAS) {
     int trustFileSet=0;
-#if 0
 # ifndef OS_WIN32
     /* try to find OpenSSL certificates */
     if (trustFileSet==0) {
@@ -502,7 +501,7 @@ int GWEN_SyncIo_Tls_Prepare(GWEN_SYNCIO
 
       nbuf=GWEN_Buffer_new(0, 256, 0, 1);
       rv=GWEN_Directory_FindFileInPaths(paths,
-                                        "ca-certificates.crt",
+                                        "ca-bundle.crt",
                                         nbuf);
       GWEN_StringList_free(paths);
       if (rv==0) {
@@ -529,6 +528,7 @@ int GWEN_SyncIo_Tls_Prepare(GWEN_SYNCIO
 # endif
 
 
+#if 0
 # ifndef OS_WIN32
     /* try to find ca-certificates (at least available on Debian systems) */
     if (trustFileSet==0) {
