--- origsrc/gwenhywfar-4.13.1/gui/qt4/Makefile.am	2014-07-23 15:28:23.000000000 -0500
+++ src/gwenhywfar-4.13.1/gui/qt4/Makefile.am	2015-10-16 01:05:38.481027300 -0500
@@ -18,6 +18,8 @@ noinst_HEADERS=\
   qt4_gui_dialog.hpp
 
 libgwengui_qt4_la_LIBADD=$(qt4_libs) $(top_builddir)/src/$(gwenhywfar_internal_libname) $(builddir)/../cpp/libgwengui-cpp.la
+libgwengui_qt4_la_LDFLAGS = -no-undefined
+
 libgwengui_qt4_la_SOURCES=\
   qt4dialogbox.cpp \
   qt4_gui.cpp \
--- origsrc/gwenhywfar-4.13.1/plugins/dbio/csv/Makefile.am	2014-07-23 15:28:23.000000000 -0500
+++ src/gwenhywfar-4.13.1/plugins/dbio/csv/Makefile.am	2015-10-16 09:11:20.120045500 -0500
@@ -22,7 +22,7 @@ csv_la_LIBADD=$(top_builddir)/src/libgwe
 
 
 csv_la_LDFLAGS = -no-undefined $(STRIPALL) \
-    -module -version-info 0:1:0
+    -module -avoid-version
 
 
 sources:
--- origsrc/gwenhywfar-4.13.1/plugins/dbio/olddb/Makefile.am	2014-07-23 15:28:23.000000000 -0500
+++ src/gwenhywfar-4.13.1/plugins/dbio/olddb/Makefile.am	2015-10-16 09:11:32.649636500 -0500
@@ -21,7 +21,7 @@ olddb_la_LIBADD=$(top_builddir)/src/libg
 
 
 olddb_la_LDFLAGS = -no-undefined $(STRIPALL) \
-    -module -version-info 0:1:0
+    -module -avoid-version
 
 sources:
 	for f in $(olddb_la_SOURCES); do \
--- origsrc/gwenhywfar-4.13.1/plugins/dbio/xmldb/Makefile.am	2014-07-23 15:28:23.000000000 -0500
+++ src/gwenhywfar-4.13.1/plugins/dbio/xmldb/Makefile.am	2015-10-16 09:11:40.029573700 -0500
@@ -21,7 +21,7 @@ xmldb_la_SOURCES=xmldb.c
 xmldb_la_LIBADD=$(top_builddir)/src/libgwenhywfar.la
 
 xmldb_la_LDFLAGS = -no-undefined $(STRIPALL) \
-    -module -version-info 0:1:0
+    -module -avoid-version
 
 sources:
 	for f in $(xmldb_la_SOURCES); do \
